src/sources/util/tcp_error.rs

```diff
@@ -0,0 +1,25 @@
+#![deny(missing_docs)]
+
+use tokio_util::codec::LinesCodecError;
+
+/// An error that occurs in the context of TCP connections.
+pub trait TcpError {
+    /// The TCP error has been fatal, the TCP stream should no longer be read
+    /// since it will hang up indefinitely.
+    ///
+    /// This can occur e.g. when reading the header of a length-delimited codec
+    /// failed and it can no longer be determined where the next header starts.
+    fn is_fatal(&self) -> bool;
```

Comment:
Would it be worth making these semantics more explicit in the function naming? I initially mistook `is_fatal` for meaning the operation could be retried. Maybe the opposite would work — `can_continue`?

Reply:
Adressed in 404e275986e7e05c54205bd8beb651239904d167